ledger-tool/src/bigtable.rs

```diff
@@ -193,25 +210,37 @@ async fn compare_blocks(
         "owned bigtable {} blocks found ",
         owned_bigtable_slots.len()
     );
-    let reference_bigtable = solana_storage_bigtable::LedgerStorage::new_with_config(ref_config)
-        .await
-        .map_err(|err| format!("failed to connect to reference bigtable: {err:?}"))?;
 
-    let reference_bigtable_slots = reference_bigtable
-        .get_confirmed_blocks(starting_slot, limit)
-        .await?;
-    info!(
-        "reference bigtable {} blocks found ",
-        reference_bigtable_slots.len(),
+    // Because the owned bigtable may include superfluous slots, stop checking
+    // the reference set at owned_bigtable_slots.last() or else the remaining
+    // reference slots will show up as missing.
+    let last_reference_block = reference_bigtable_slots
+        .last()
+        .expect("already returned if reference_bigtable_slots is empty");
+    let last_block_checked = owned_bigtable_slots
+        .last()
+        .map(|last_owned_block| min(last_owned_block, last_reference_block))
+        .unwrap_or(last_reference_block);
```

Comment:
nit: What do you think about moving this logic inside `missing_slots()` and spitting it out in `MissingBlocksData` struct ? Given that the value can be derived from the two lists, it is arguable that passing this separately is a redundant arg. Deriving it internally in that function also protects against possible errors (such as passing a subset of the vecs into `missing_blocks()`)

Reply:
Yep, good call!

Reply:
Okay, pushed as 3 new commits. No other changes; just rebased on master to make sure nothing stale.